test/plots/xyPlotTests.ts

```diff
@@ -2,268 +2,348 @@
+        const dataset = new Plottable.Dataset(testData);
+        dataset.data().forEach((d, i) => {
+          assert.strictEqual(plot.x().accessor(d, i, dataset), accessor(d), "returns the input accessor");
+        });
+      });
+
+      it("can set the x property be based on scaled data", () => {
+        const accessor = (d: any) => d * 10;
+        const scale = new Plottable.Scales.Linear();
+        assert.strictEqual(plot.x(accessor, scale), plot, "setting the x property returns the plot");
+
+        const testData = [4, 2, 3, 5, 6];
+        const dataset = new Plottable.Dataset(testData);
+        dataset.data().forEach((d, i) => {
+          const bindingAccessor = plot.x().accessor;
+          const bindingScale = plot.x().scale;
+          assert.strictEqual(bindingScale.scale(bindingAccessor(d, i, dataset)),
+            scale.scale(accessor(d)), "returns the input accessor and scale");
+        });
+      });
+
+      it("sets the range on the input scale if the plot has a width", () => {
+        const svg = TestMethods.generateSVG();
+        plot.anchor(svg);
+        plot.computeLayout();
+
+        const scale = new Plottable.Scales.Linear();
+        assert.strictEqual(plot.x(0, scale), plot, "setting the x property returns the plot");
+
+        assert.deepEqual(scale.range(), [0, plot.width()], "range goes to width on scale");
+
         svg.remove();
       });
 
-      it("deferredRendering() setter/getter", () => {
-        assert.strictEqual(plot.deferredRendering(), false, "deferred rendering is false by default");
-        assert.strictEqual(plot.deferredRendering(true), plot, "setting the deferred rendering option returns the plot");
-        assert.strictEqual(plot.deferredRendering(), true, "deferred rendering can be turned on");
-        plot.deferredRendering(false);
-        assert.strictEqual(plot.deferredRendering(), false, "deferred rendering can be turned off");
+      it("can install its extent onto the input scale", () => {
+        const svg = TestMethods.generateSVG();
+        const data = [0, 1, 2, 3, 4];
+        plot.addDataset(new Plottable.Dataset(data));
+
+        plot.anchor(svg);
+
+        const scale = new Plottable.Scales.Linear();
+        scale.padProportion(0);
+        assert.strictEqual(plot.x((d) => d, scale), plot, "setting the x property returns the plot");
+
+        const mathUtils = Plottable.Utils.Math;
+        assert.deepEqual(scale.domain(), [mathUtils.min(data, 0), mathUtils.max(data, 0)], "range goes to width on scale");
+
+        svg.remove();
+      });
+    });
+
+    describe("managing the y property", () => {
+      let plot: Plottable.XYPlot<number, number>;
+
+      beforeEach(() => {
+        plot = new Plottable.XYPlot<number, number>();
+
+        // HACKHACK: Must install y scale https://github.com/palantir/plottable/issues/2934
+        plot.x(0, new Plottable.Scales.Linear());
+      });
+
+      afterEach(() => {
+        plot.destroy();
```

Comment:
nvm, I made a comment regarding plot.destroy in previous describe block and removed it when I realized it is fine; I guess I forgot to remove this one :p